api/app/clients/tools/structured/StableDiffusion.js

```diff
@@ -61,15 +61,27 @@ class StableDiffusionAPI extends StructuredTool {
   async _call(data) {
     const url = this.url;
     const { prompt, negative_prompt } = data;
+    let defaultPayload;
+    try {
+      defaultPayload = process.env.SD_WEBUI_DEFAULT_PARAMETERS
+        ? JSON.parse(process.env.SD_WEBUI_DEFAULT_PARAMETERS)
+        : {
+            sampler_name: 'DPM++ 2M Karras',
+            cfg_scale: 4.5,
+            steps: 22,
+            width: 1024,
+            height: 1024,
+          };
+    } catch (error) {
+      logger.error('[StableDiffusion] Error while parsing JSON:', error);
+    }
+
     const payload = {
-      prompt,
-      negative_prompt,
-      sampler_index: 'DPM++ 2M Karras',
-      cfg_scale: 4.5,
-      steps: 22,
-      width: 1024,
-      height: 1024,
+      prompt: prompt,
+      negative_prompt: defaultPayload.negative_prompt || negative_prompt,
```

Comment:
This will throw an error if you run into a parsing error.

Reply:
I added an additional try catch for this, plus set the default params in case of a bad JSON